test/Atc.Cosmos.EventStore.Cqrs.Tests/InMemoryStoreV2.cs

```diff
@@ -0,0 +1,241 @@
+﻿using System.Runtime.CompilerServices;
+using Atc.Cosmos.EventStore.Cosmos;
+using Atc.Cosmos.EventStore.Streams;
+using Microsoft.Azure.Cosmos;
+
+namespace Atc.Cosmos.EventStore.Cqrs.Tests;
+
+/// <summary>
+/// A working in-memory implementation of Atc.Cosmos.EventStore.IEventStoreClient.
+/// </summary>
+public sealed class InMemoryEventStoreClient(IStreamReadValidator readValidator) : IEventStoreClient
```

Comment:
I like your implementation and the options it give us 👍 
The implementation should replace the existing `InMemoryEventStore.cs` as it has never been used.